src/org/mozilla/javascript/ThreadSafeSlotMapContainer.java

```diff
@@ -62,13 +62,21 @@ public boolean isEmpty() {
     }
 
     @Override
-    public Slot get(Object key, int index, SlotAccess accessType) {
+    public Slot modify(Object key, int index, int attributes) {
+        final long stamp = lock.writeLock();
+        try {
+            checkMapSize();
```

Comment:
It's probably fine to call checkMapSize here, but should the implementation of checkMapSize in this class actually do anything? Calling super doesn't seem right because at best it won't do anything (current behavior,) and at worst it could swap the map for a non-threadsafe version.

Reply:
SlotMapContainer and ThreadSafeSlotMapContainer are just wrappers for either an EmbeddedSlotMap or a HashSlotMap. Neither of those last two types are thread safe -- we're relying on the thread-safe wrapper to do the locking. So there isn't any chance of the map on a particular object being replaced with a non-thread safe version.